### ISSUE TITLE
Errors logged at console are not logged when running as container

### ISSUE THREAD

**Issue by jtackaberry** [minio/minio#19125]
When starting minio directly from the command line, startup errors are clearly logged.  For example, here's one with a misconfigured DNS name:

```
$ ./minio -version
minio version RELEASE.2024-02-24T17-11-14Z (commit-id=c2b54d92f66ea02642be4ede26f2b7ecac1f2056)
Runtime: go1.21.7 linux/amd64
License: GNU AGPLv3 <https://www.gnu.org/licenses/agpl-3.0.html>
Copyright: 2015-2024 MinIO, Inc.

$ ./minio server http://bogus-{0...4}.example.com:9000/export
ERROR Invalid command line arguments: lookup bogus-0.example.com on 127.0.0.53:53: no such host
      > Please provide correct combination of local/remote paths
      HINT:
        For more information, please refer to https://min.io/docs/minio/linux/operations/concepts/erasure-coding.html
```

Very clear what went wrong here. 

However, when running under docker, no error is logged:

```
$ docker run -ti minio/minio:RELEASE.2024-02-24T17-11-14Z server http://bogus-{0...4}.example.com:9000/export
[no text]
```

## Expected Behavior

The DNS lookup error should also be logged when Minio is running as a container.

## Current Behavior

When running as a container, Minio logs nothing, and (more incidentally) doesn't terminate.  The non-terminating behavior does make sense to me, as it's being run as a server, although the lack of error logging turned this trivial configuration error into a protracted troubleshooting session.

## Steps to Reproduce (for bugs)

1. Run `docker run -ti minio/minio:RELEASE.2024-02-24T17-11-14Z server http://bogus-{0...4}.example.com:9000/export`

## Context

I was building a Minio cluster as a StatefulSet in Kubernetes and typoed the node FQDN.  

## Your Environment

Minio `RELEASE.2024-02-24T17-11-14Z`

Above example ran on Ubuntu 22.04 on amd64.  Also reproduced in Docker and containerd on Linux amd64.

**Comment by harshavardhana** [comment]
Will take a look. It must some terminal issue in containers.

**Comment by harshavardhana** [comment]
This looks like a bug it was not intentional to avoid logging; we expected it to log. Sent a fix that would work here #19129
